clang/lib/Sema/SemaInit.cpp

```diff
@@ -8238,6 +8246,15 @@ ExprResult InitializationSequence::Perform(Sema &S,
         }
       }
 
+      // Note the return value isn't used to return early so that additional
+      // diagnostics can be emitted and to preserve the AST as best as possible
+      // even though an error might have occurred. For struct initialization it
+      // also allows all field assignments to be checked rather than bailing on
+      // the first error.
+      (void)S.BoundsSafetyCheckInitialization(
```

Comment:
Same here. The cast to `(void)` also isn’t necessary I’d say.

Reply:
The comment is explaining why we aren't checking the return value. If you remove it isn't obvious at all why its being ignored.